File: OCH_Tao/ABC/B/306.py
print(int("".join(input().split())[::-1], base=2))

# A = list(map(lambda x: bool(int(x)), input().split()))
# cnt = 0
# for i in range(64):
#   if A[i]:
#     cnt += 2**i
# print(cnt)

# A = list(map(int, input().split()))
# print(sum([2**i*j for i, j in enumerate(A)]))

File: OCH_Tao/ABC/B/306.js
/*
const A = require("fs").readFileSync("/dev/stdin", "utf8").trim().split(" ").map(x => parseInt(x, 10));
let cnt = 0n;
for (const [i, j] of A.entries()) {
  cnt += 2n ** BigInt(i) * BigInt(j);
}
console.log(cnt.toString());
*/

console.log(BigInt("0b" + require("fs").readFileSync("/dev/stdin", "utf8").trim().split(" ").reverse().join("")).toString());